src/frequenz/client/reporting/_client.py

```diff
@@ -157,6 +157,41 @@ async def list_single_component_data(
             for entry in page:
                 yield entry
 
+    # pylint: disable=too-many-arguments
+    async def list_multiple_microgrid_components_data(
+        self,
+        *,
+        microgrid_components: list[tuple[int, list[int]]],
+        metrics: Metric | list[Metric],
+        start_dt: datetime,
+        end_dt: datetime,
+        page_size: int = 1000,
+    ) -> AsyncIterator[MetricSample]:
+        """Iterate over the data for multiple microgrids and components.
+
+        Args:
+            microgrid_components: List of tuples where each tuple contains
+                                  microgrid ID and corresponding component IDs.
+            metrics: The metric name or list of metric names.
+            start_dt: The start date and time.
+            end_dt: The end date and time.
+            page_size: The page size.
+
+        Yields:
+            A named tuple with the following fields:
+            * timestamp: The timestamp of the metric sample.
+            * value: The metric value.
```

Comment:
The documentation should be like [here](https://github.com/frequenz-floss/frequenz-client-reporting-python/pull/39/files#diff-bc44ef8bfdafafc5ae638f3475097e7459739c5eda848f6954aa620b48f2e50aL73)